src/index.ts

```diff
@@ -149,14 +152,38 @@ export async function main(
       coverageFilePaths = args.file
     }
   }
-  coverageFilePaths = coverageFilePaths.concat(await getCoverageFiles(
-    args.dir || projectRoot,
-    coverageFilePaths.length > 0 ? coverageFilePaths : coverageFilePatterns(),
-  ))
+
+let searchPath = projectRoot
+
+// If customer provided an exclude to -s
+if (args.dir) {
+    if (args.dir.startsWith('!') || args.dir.startsWith('^')) {
+      verbose(`Adding search path to ignore list: ${args.dir}`)
```

Comment:
When ran with `-s ^test,coverage.\*` it gets added to the ignore list as `test,coverage.*` as I would expect. However, we still locate files named coverage.*`

What am I missing here, @mitchell-codecov ?